dev/com.ibm.ws.security.token.ltpa/src/com/ibm/ws/security/token/ltpa/internal/LTPAConfigurationImpl.java

```diff
@@ -156,18 +158,34 @@ private void loadConfig(Map<String, Object> props) {
         // expirationDifferenceAllowed is set to 3 seconds (3000ms) by default.
         // If expirationDifferenceAllowed is set to less than 0, then the two expiration values will not be compared in the LTPAToken2.decrypt() method.
         expirationDifferenceAllowed = (Long) props.get(KEY_EXP_DIFF_ALLOWED);
-        monitorDirectory = (Boolean) props.get(CFG_KEY_MONITOR_DIRECTORY);
-
-        resolveActualPrimaryKeysFileLocation();
+        enableDirectoryMonitoring = (Boolean) props.get(CFG_KEY_ENABLE_DIRECTORY_MONITORING);
+        updateTrigger = (String) props.get(CFG_KEY_UPDATE_TRIGGER);
 
         //get all validationKeys elements
         Map<String, List<Map<String, Object>>> validationKeysElements = Nester.nest(props, CFG_KEY_VALIDATION_KEYS);
         if (!validationKeysElements.isEmpty()) {
+            isValidationKeysFileConfigured = !validationKeysElements.get(CFG_KEY_VALIDATION_KEYS).isEmpty();
+        }
+
+        resolveActualPrimaryKeysFileLocation();
+
+        if (isValidationKeysFileConfigured) {
             configValidationKeys = getConfigValidationKeys(validationKeysElements, CFG_KEY_VALIDATION_KEYS, CFG_KEY_VALIDATION_FILE_NAME, CFG_KEY_VALIDATION_PASSWORD,
-                                                           CFG_KEY_VALIDATION_NOT_USE_AFTER_DATE);
+                                                           CFG_KEY_VALIDATION_VALID_UNTIL_DATE);
+        } else {
+            configValidationKeys = null;
         }
 
-        if (monitorDirectory) {
+        if (updateTrigger != null && !updateTrigger.equalsIgnoreCase("polled")){
+            if (enableDirectoryMonitoring) {
+            Tr.warning(tc, "LTPA_UPDATE_TRIGGER_NOT_POLLED_AND_MONITOR_DIRECTORY_TRUE", enableDirectoryMonitoring);
+            }
+            if (monitorInterval > 0) {
+                Tr.warning(tc, "LTPA_UPDATE_TRIGGER_NOT_POLLED_AND_MONITOR_INTERVAL_GREATER_THAN_ZERO", monitorInterval);
+            }
+        }
+
+        if (enableDirectoryMonitoring) {
             if (monitorInterval <= 0) {
                 Tr.warning(tc, "LTPA_MONITOR_DIRECTORY_TRUE_AND_FILE_MONITOR_NOT_ENABLED", monitorInterval);
```

Comment:
With all the changes, we probably should have MONITOR_INTERVAL is equals or less than zero instead of MONITOR_NOT_ENABLE